packages/ember-data/lib/system/store/serializer-response.js

```diff
@@ -3,6 +3,35 @@ import Model from 'ember-data/system/model/model';
 const get = Ember.get;
 
 /**
+  This is a helper method that validates a JSON API top-level document
+
+  The format of a document is described here:
+  http://jsonapi.org/format/#document-top-level
+
+  @method isValidJsonApiDocument
+  @param {Object} doc
+  @return {Boolean} true if the document is valid, false otherwise
+*/
+export function hasJsonApiDocumentValidationErrors(doc) {
+  let errors = [];
+  if (!doc || typeof doc !== 'object') {
+    errors.push('Top level of a JSON API document must be an object');
+  } else {
+    if (!('data' in doc) &&
+        !('errors' in doc) &&
+        !('meta' in doc)) {
+      errors.push('One or more of the following keys must be present: "data", "errors", "meta".');
+    } else {
+      if (('data' in doc) && ('errors' in doc)) {
```

Comment:
arrays or objects are allowed as values. I'm not sure where we want this to end, because we could then validate the objects (or arrays) themselves, including relationships, etc...